### DISCUSSION TITLE
hashi_vault is not recognizing Hashicorp Vault Environment variables

### DISCUSSION THREAD

**Discussion by madhu3984** [ansible-collections/community.hashi_vault#372] [Q&A] [answered]
Hi Team, I am testing secrets retrieval from my Hashicorp vault system using Ansible and using hashi_vault module/plugin/lookup filter.

My ansible version is core 2.14.4 and python version is 3.9.13

I set my Hashicorp details as environment variables. 

export VAULT_ADDR="XXXX"
export VAULT_AUTH_METHOD="approle"
export VAULT_ROLE_ID="YYYY"
export VAULT_SECRET_ID="ZZZZ"

When I run the playbook, I am getting below error.

TASK [Authenticate with a vault app role and retrieve secrets under path secret/uat/ansible/secrets] *********************************************************************
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'hashi_vault'. Error was a <class 'ansible.errors.AnsibleError'>, original message: No Vault Token specified or discovered.. No Vault Token specified or discovered."}

Come some one please help me on this?

Thanks in advance.

Regards,
Madhu M

**Comment by madhu3984** [comment]
I found out that i have to use new ENV variables 

ANSIBLE_HASHI_VAULT_AUTH_METHOD
ANSIBLE_VAULT_ROLE_ID
ANSIBLE_HASHI_VAULT_ROLE_ID
ANSIBLE_VAULT_ADDR

After adding this, its working fine.

**Reply by ji-podhead** [reply]
so you have to use specific naming convention for your variables for this to work?
i also noticed that...

**Comment by briantist** [comment, marked as answer]
Some environment variables used by the Vault CLI are accepted, mostly for historical reasons, or because they are used implicitly. 

But we generally prefer ansible-specific vars and discourage use of the Vault ones. More info:
- https://github.com/ansible-collections/community.hashi_vault/issues/10
